src/storage.h

```diff
@@ -0,0 +1,273 @@
+/// The Storage class inherits from the Facility class and is
+/// dynamically loaded by the Agent class when requested.
+///
+/// @section intro Introduction
+/// This Agent was initially developed to support the fco code-to-code 
+/// comparsion.
+/// It's very similar to the "NullFacility" of years 
+/// past. Its purpose is to convert a commodity from one commodity to another 
+/// after some period of delay time. This facility is very good for use as a 
+/// storage facility or fuel fabrication.
+///
+/// @section agentparams Agent Parameters
+/// in_commod is a string naming the commodity that this facility recieves
+/// out_commod is a string naming the commodity that in_commod is stocks into
+/// process_time is the number of timesteps between receiving and offering
+/// in_recipe (optional) describes the incoming resource by recipe
+/// out_recipe (optional) describes the outgoing resource by recipe
+/// 
+/// @section optionalparams Optional Parameters
+/// max_inv_size is the maximum capacity of the inventory storage
+/// capacity is the maximum processing capacity per timestep
+///
+/// @section detailed Detailed Behavior
+/// 
+/// Tick:
+/// Nothing really happens on the tick. 
+///
+/// Tock:
+/// On the tock, any material that has been waiting for long enough (delay 
+/// time) is stocks and placed in the stocks buffer.
+///
+/// Any brand new inventory that was received in this timestep is placed into 
+/// the processing queue to begin waiting. 
+/// 
+/// Making Requests:
+/// This facility requests all of the in_commod that it can.
+///
+/// Receiving Resources:
+/// Anything of the in_commod that is received by this facility goes into the 
+/// inventory.
+///
+/// Making Offers:
+/// Any stocks material in the stocks buffer is offered to the market.
+///
+/// Sending Resources:
+/// Matched resources are sent immediately.
+class Storage 
+  : public cyclus::Facility,
+    public cyclus::toolkit::CommodityProducer {
+ public:  
+  /// Constructor for Storage Class
+  /// @param ctx the cyclus context for access to simulation-wide parameters
+  Storage(cyclus::Context* ctx);
+
+  /// The Prime Directive
+  /// Generates code that handles all input file reading and restart operations
+  /// (e.g., reading from the database, instantiating a new object, etc.).
+  /// @warning The Prime Directive must have a space before it! (A fix will be
+  /// in 2.0 ^TM)
+  
+  #pragma cyclus decl
+
+  #pragma cyclus note {"doc": "A storage facility converts from one " \
+                              "commodity to another, with an optional delay."}
+
+  /// A verbose printer for the Storage
+  virtual std::string str();
+
+  // --- Facility Members ---
+  
+  // --- Agent Members ---
+  virtual void EnterNotify();
+
+  /// The handleTick function specific to the Storage.
+  /// @param time the time of the tick  
+  virtual void Tick();
+
+  /// The handleTick function specific to the Storage.
+  /// @param time the time of the tock
```

Comment:
This param line is an artifact of a previous implementation of tock, I think. Tock no longer accepts time as a parameter.